### PR TITLE
Draft of skribo kickoff post

### PR DESCRIPTION
**Pull request by raphlinus**

### CODE REVIEW

File: _posts/2019-02-25-text-layout-kickoff.markdown [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+layout: post
+title:  "Low level text layout kickoff"
+date:   2019-02-27 11:16:42 -0700
+categories: [rust, skribo, text]
+---
+This post is to announce a new Rust library for low-level text layout, called "[skribo]" (the Esperanto word for "writing"). This has been a major gap in the Rust ecosystem, and I hope the new crate can improve text handling across the board.
+
+## Multiple shaping engines
+
+The skribo library doesn't do shaping itself; it is basically a glue library on top of an actual shaping engine. Most users will use [HarfBuzz] for this, the industry-standard library. However, a goal for skribo is that it abstracts over the details of the shaping engine. Other choices include platform text (DirectWrite on Windows, CoreText on macOS and iOS). Also, if and when a Rust-native shaping library emerges, we want to make it easy to switch to that.
```
Comment (cmyr) [inline _posts/2019-02-25-text-layout-kickoff.markdown]: I would briefly explain what shaping is before diving in here?

File: _posts/2019-02-25-text-layout-kickoff.markdown [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+The choice of whether to use HarfBuzz or platform shaping is a tradeoff. Using HarfBuzz means more consistency across platforms, and thus an easier testing story. Using platform text means more consistency with other apps on the platform, and less code in the critical path (so shorter compile times and smaller executables).
+
+## Expected users
+
+This work is being funded by Mozilla Research to be used in Servo. Doing text layout for the Web is a very complex problem. The higher level representation of rich text is very tightly bound to the DOM and CSS. All that logic needs to live in Servo, with the lower level providing a clean interface to lay out a span of text with a *single* style and a single [BiDi] direction.
+
+The second major use case is [piet], the 2D graphics abstraction I've been working on. Currently for the Cairo back-end piet uses the "toy text API," which is of course [inadequate][piet#10]. The idea is that skribo handles text layout for Cairo and all future low-level drawing back-ends (including any future back-end using WebRender or direct Vulkan drawing).
+
```
Comment (cmyr) [inline _posts/2019-02-25-text-layout-kickoff.markdown]: Not necessary, but another concrete use case would be text measurement in xi.

File: _posts/2019-02-25-text-layout-kickoff.markdown [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+Another major problem is making sure locale information affects layout properly. One of the most important to solve is [Han unification], which is the fact that Chinese, Japanese, and Korean share unicode codepoints even though they can be considered different scripts. Often, they should be rendered with different fonts (meaning that Unicode coverage is not the only criterion for selecting fonts). Alternatively, fonts like [Source Han Sans] have variant glyphs for all CJK languages, and use the "locl" OpenType feature to select them. A good layout library handles both, transparently to applications.
+
+Other problems within scope are figuring out logic for "fake bold" and "fake italic" when the font collection doesn't provide true versions and adding [letter-spacing].
```
Comment (cmyr) [inline _posts/2019-02-25-text-layout-kickoff.markdown]: hmm, I would ~versions and~ -> "versions, as well as ..."; and seems ambiguous in this context (is it an additional thing that the font collection doesn't provide?)

File: _posts/2019-02-25-text-layout-kickoff.markdown [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+Looking forward, [font variations] affect layout, so one of the goals is to plumb this through to the underlying shaping engine, as supported.
```
Comment (cmyr) [inline _posts/2019-02-25-text-layout-kickoff.markdown]: as supported/where supported? 🤷‍♂️

File: _posts/2019-02-25-text-layout-kickoff.markdown (hunk 1 above)
Comment (raphlinus) [reply _posts/2019-02-25-text-layout-kickoff.markdown]: Better I think is to give a visual example of with/without. I'm making that now.

**Pull request merged**